server/routerlicious/Dockerfile

```diff
@@ -1,7 +1,12 @@
 # Copyright (c) Microsoft Corporation and contributors. All rights reserved.
 # Licensed under the MIT License.
 
-FROM node:14.19.1-stretch-slim AS base
+# Override this by setting the REGISTRY argument when calling docker build
+# to use a different registry than docker hub
+# docker build . REGISTRY=<custom registry>
+ARG REGISTRY=docker.io
+
+FROM ${REGISTRY}/node:14-slim AS base
```

Comment:
Another thing is "stretch-slim" vs "slim". I [just learned](https://medium.com/swlh/alpine-slim-stretch-buster-jessie-bullseye-bookworm-what-are-the-differences-in-docker-62171ed4531d) that "stretch" refers to the Debian version 9, and the "de-facto" Node.js image like "node:14" or even "node:14-slim" might use the latest Debian version.

It might not make a big difference, but I wonder if there is any behavior in Debian Stretch vs. latest that could cause differences in our service's behavior. I sent a request to the Supply Chain team to add "stretch-slim" to the list of golden images, which would become available in MCR. I am still waiting for their response.

Reply:
yeah that's a good question indeed
I'm not sure
If they can get stretch into MCR that would be the safest option 👍

Reply:
Again, here we specify stretch-slim to resolve underlying security vulnerabilities in Debian or any of the versions of various linux packages being pulled in. It would be counter-productive to regress our version to a version with security vulnerabilities in its underlying packages. Functionality should be the same, but security is the concern here.